src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,40 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(bitten) {
+    if (bitten.hidden === false) {
+      bitten.health -= 50;
+    }
+
+    if (bitten.health <= 0) {
+      const indexOfDead = Animal.alive.indexOf(bitten);
+
+      Animal.alive.splice(indexOfDead, 1);
+    }
+  }
```

Comment:
In the `bite` method, consider adding a check to ensure that the `bitten` object is an instance of `Herbivore`. This will prevent errors if a `Carnivore` accidentally tries to bite another `Carnivore` or a non-Herbivore object.